packages/core/src/Renderer.ts

```diff
@@ -79,6 +97,146 @@ export abstract class Renderer extends Component {
     this.shaderData._addRefCount(1);
   }
 
+  /**
+   * Get the first instance material.
+   * @returns The first instance material
+   */
+  getInstanceMaterial(): Material | null;
+
+  /**
+   * Get the first instance material by index.
+   * @remarks Calling this function for the first time after the material is set will create an instance material to ensure that it is unique to the renderer.
+   * @param index - Material index
+   * @returns Instance material
+   */
+  getInstanceMaterial(index: number): Material | null;
+
+  getInstanceMaterial(index: number = 0): Material | null {
+    const materials = this._materials;
+    if (materials.length > index) {
+      const material = materials[index];
+      if (material) {
+        if (this._materialsInstanced[index]) {
+          return material;
+        } else {
+          return this._createInstanceMaterial(material, index);
+        }
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Get the first material.
+   * @returns The first material
+   */
+  getMaterial(): Material | null;
+
+  /**
+   * Get the first material by index.
+   * @param index - Material index
+   * @returns Material
+   */
+  getMaterial(index: number): Material | null;
+
+  getMaterial(index: number = 0): Material | null {
+    const materials = this._materials;
+    return materials.length > 0 ? materials[index] : null;
+  }
+
+  /**
+   * Set the first material.
+   * @param material - The first material
+   */
+  setMaterial(material: Material);
+
+  /**
+   * Set material by index.
+   * @param index - Material index
+   * @param material - The material
+   */
+  setMaterial(index: number, material: Material);
+
+  setMaterial(indexOrMaterial: number | Material, material: Material = null) {
```

Comment:
Can use setMaterial(material: Material, indexOfMaterial: number = 0)

Reply:
Not Good! If no function overload, I will design like that. If use function overload,this is better for developers.